storage/heap/CMakeLists.txt

```diff
@@ -26,8 +26,12 @@ SET(HEAP_SOURCES  _check.c _rectest.c hp_block.c hp_clear.c hp_close.c hp_create
 MYSQL_ADD_PLUGIN(heap ${HEAP_SOURCES} STORAGE_ENGINE MANDATORY
                  RECOMPILE_FOR_EMBEDDED DTRACE_INSTRUMENTED)
 
-ADD_EXECUTABLE(hp_test1 hp_test1.c)
-TARGET_LINK_LIBRARIES(hp_test1 mysys heap dbug strings)
+IF(WITH_UNIT_TESTS AND WITH_EMBEDDED_SERVER)
```

Comment:
Does this mean that hp_tests will not get built in case of -DWITH_EMBEDDED_SERVER=OFF ?

Reply:
Yes, this is backported from 5.7

Reply:
Consider reporting upstream (although it might have been already fixed due to 8.0 removing embedded server)